src/screens/MainScreen.tsx

```diff
@@ -1,26 +1,56 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import StrollScreen from "./StrollScreen";
 import MapScreen from "./MapScreen";
 import DetailedViewScreen from "./DetailedViewScreen";
+import LogInScreen from "./LoginScreen";
+import CreateAccountScreen from "./CreateAccountScreen";
+import NotLoggedIn from "./NotLoggedIn";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+export enum Screens {
+	Home = "mapScreen",
+	Stroll = "strollScreen",
+	Details = "detailsScreen",
+	NotLoggedIn = "notLoggedIn",
+	Login = "loginScreen",
+	CreateAccount = "createAccountScreen",
+}
 
 /**
  * Serves as the main view of the app, here all
  * of the other screens are switched between.
  * @returns the main screen component
  */
 const MainScreen: React.FC = () => {
-	const [page, setPage] = useState("mapScreen");
+	const [token, setToken] = useState(false);
+
+	useEffect(() => {
+		//TODO this doesn't work?? idk why, please help
+		if (AsyncStorage.getItem("@auth_token") !== null) {
+			setToken(true);
+			return;
+		}
+		setToken(false);
+	}, []);
+
+	const [page, setPage] = useState<Screens>(token ? Screens.Home : Screens.NotLoggedIn);
```

Comment:
We should discuss whether we want to require logins to view the app, like this says, or just require logins to leave reviews. 

We formerly decided just require logins to leave reviews, but in a real world setting it might lead to people not leaving reviews bc they don't want to create an account if they can already access the app without it.

Reply:
I'm still pro requiring login only for leaving reviews, because if someone is already not interested in leaving reviews then they'll just use the app without leaving reviews, regardless if they are logged in or not, at least in my experience

for example, I'm logged into steam all the time, but I don't leave reviews on every game I play

Reply:
Just so this is documented, as per our conversation the other day, I will be implementing the login-before-you-can-do-anything solution for now. We can always change this later as discussed.